### PR TITLE
fix(persist-state): added missing export "MaybeAsync" type

### PR DESCRIPTION
**Pull request by stLmpp**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/elf/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

MaybeAsync type is not exported from @ngneat/elf-persist-state

## What is the new behavior?

MaybeAsync is exported from @ngneat/elf-persist-state

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

It might be useful for a custom StateStorage, since it's required to return this type in the implementation

### CODE REVIEW

**Comment by NetanelBasal** [comment]
Why do you need it?

**Comment by stLmpp** [comment]
Well, since the interface StateStorage is like this:

```typescript
export interface StateStorage {
  getItem<T extends Record<string, any>>(
    key: string
  ): MaybeAsync<T | undefined>;

  setItem(key: string, value: Record<string, any>): MaybeAsync<boolean>;

  removeItem(key: string): MaybeAsync<boolean>;
}
```

I imagine the implementation of a custom StateStorage being somethings like this:

```typescript
import { StateStorage, MaybeAsync } from '@ngneat/elf-persist-state'

export class CustomStateStorage implements StateStorage  {

  getItem<T extends Record<string, any>>(
    key: string
  ): MaybeAsync<T | undefined> {
    // My custom logic here
  }

  setItem(key: string, value: Record<string, any>): MaybeAsync<boolean> {
    // My custom logic here
  }

  removeItem(key: string): MaybeAsync<boolean> {
    // My custom logic here
  }  
}

export const customStateStorage = new CustomStateStorage();
// In the application I'm creating it's a little more complex than this, I have some parameters that goes
// in the constructor, and they're dynamic based on the store
```

So, it would be required as a return type if this eslint is applied for example: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-function-return-type.md

File: packages/persist-state/src/lib/storage.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Observable, of } from 'rxjs';
 
-type MaybeAsync<T> = Promise<T> | Observable<T>;
+export type MaybeAsync<T> = Promise<T> | Observable<T>;
```
Comment (NetanelBasal) [inline packages/persist-state/src/lib/storage.ts:3]: Can you please change it to Async<T>? it's not maybe :)

Comment (stLmpp) [reply packages/persist-state/src/lib/storage.ts]: Done

**Pull request merged**